lib/path/default.nix

```diff
@@ -271,6 +271,46 @@ in /* No rec! Add dependencies on this file at the top. */ {
               second argument: "${toString path2}" with root "${toString path2Deconstructed.root}"'';
         joinRelPath components;
 
+  /*
+  Split the filesystem root from a path.
+  The result is an attribute set with these attributes:
+
+  - `root`: The filesystem root of the path, meaning that this directory has no parent directory.
+  - `subpath`: The normalised subpath string that leads to the given path from the filesystem root, see `lib.path.subpath.normalise`
+
+  Laws:
+
+  - Appending the `root` and `subpath` gives the original path:
+
+        append (splitRoot p).root (splitRoot p).subpath == p
+
+  - Trying to get the parent directory of `root` using `readDir` returns `root` itself:
+
+        dirOf (splitRoot p).root == (splitRoot p).root
+
+  Type:
+    splitRoot :: Path -> { root :: Path, subpath :: String }
+
+  Example:
+    splitRoot /foo/bar
+    => { root = /.; subpath = "./foo/bar"; }
+    splitRoot /.
+    => { root = /.; subpath = "./."; }
+    # Nix neutralises `..` path components for all paths automatically
+    splitRoot /foo/../bar
+    => { root = /.; subpath = "./bar"; }
+  */
```

Comment:
Please also refer to the `deconstructPath` function in the doc and vice versa.

Reply:
That's only an internal function though